Update the docs.

vpetrovykh opened the pull request:
Move things from datamodel to standard library. Update some of the
descriptive text.

1st1 commented on docs/datamodel/functions.rst:
```diff
@@ -18,17 +18,39 @@ of :eql:type:`int64`:
+arguments, a cross-product of all the input sets is computed to
+determine all the input combinations that the function needs to be
+applied to. After that, the function is applied to each element of the
+cross-product in turn to generate the output set. This implies that if
```
"in turn" -> "in order"? this sentence is a bit hard to read.

vpetrovykh replied:
This is hard to word right. The "in turn" here is referring to "the function is applied to each element". The order doesn't matter and it's just a way of saying "apply element-wise".

If I change it to "in order" it will look like it refers to "in order to generate the output set".

I think both of these are about the same level of awkward wording, so I've tried splitting the sentence instead.

1st1 commented on docs/stdlib/datetime.rst:
```diff
@@ -5,7 +5,263 @@
+.. _ref_std_datetime_ops:
+
+=======================
+Functions and Operators
```
Hm, I'm not sure it's a good idea to push the func/ops table so far down. How about we move the table to the very top and make the first few rows to link to the actual types? The types description would be right below the table, which essentially is a TOC of the page.

vpetrovykh replied:
ok

1st1 commented on docs/stdlib/json.rst:
```diff
@@ -4,7 +4,120 @@
+
+
+=======================
+Functions and Operators
```
Ditto, the toc table should stay on top of the page.

vpetrovykh replied:
ok

1st1 commented on docs/stdlib/numerics.rst:
```diff
@@ -4,7 +4,224 @@
+
+
+====================================================
+Arithmetic Operators and Numeric Converter Functions
```
ditto

vpetrovykh replied:
ok

1st1 reviewed (changes requested):
Overall this looks awesome, but I think we need to move all type descriptions under the TOC table.

vpetrovykh commented:
I've moved the func/ops tables up to be a sort of TOC for all the pages in the standard library.

I've also added `BaseObject` and `Object` to stdlib docs.
